pinot-segment-local/src/test/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/RealtimeNativeTextIndexConcurrentTest.java

```diff
@@ -0,0 +1,306 @@
+      throws InterruptedException, IOException {
+    CountDownLatch countDownLatch = new CountDownLatch(2);
+    List<String> words = new ArrayList<>();
+    words.add("ab");
+    words.add("abba");
+    words.add("aba");
+    words.add("bab");
+    words.add("cdd");
+    words.add("efg");
+
+    try (RealtimeNativeTextIndex textIndex = new RealtimeNativeTextIndex("testFSTColumn")) {
+      _threadPool.submit(() -> {
+        try {
+          performReads(textIndex, words, 10, 200, countDownLatch);
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      });
+
+      _threadPool.submit(() -> {
+        try {
+          performWrites(textIndex, words, 10, countDownLatch);
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      });
+
+      countDownLatch.await();
+    }
+
+    assertEquals(_resultSet.size(), words.size());
+
+    assertTrue("ab not found in result set", _resultSet.contains("ab"));
+    assertTrue("abba not found in result set", _resultSet.contains("abba"));
+    assertTrue("aba not found in result set", _resultSet.contains("aba"));
+    assertTrue("bab not found in result set", _resultSet.contains("bab"));
+    assertTrue("cdd not found in result set", _resultSet.contains("cdd"));
+    assertTrue("efg not found in result set", _resultSet.contains("efg"));
+  }
+
+  @Test
+  public void testConcurrentWriteWithMultipleThreads()
+      throws InterruptedException, IOException {
+    List<String> firstThreadWords = new ArrayList<>();
+    List<String> secondThreadWords = new ArrayList<>();
+    List<String> mergedThreadWords = new ArrayList<>();
+    CountDownLatch countDownLatch = new CountDownLatch(3);
+    firstThreadWords.add("ab");
+    firstThreadWords.add("abba");
+    firstThreadWords.add("aba");
+    secondThreadWords.add("bab");
+    secondThreadWords.add("cdd");
+    secondThreadWords.add("efg");
+
+    mergedThreadWords.addAll(firstThreadWords);
+    mergedThreadWords.addAll(secondThreadWords);
+
+    try (RealtimeNativeTextIndex textIndex = new RealtimeNativeTextIndex("testFSTColumn")) {
+      _threadPool.submit(() -> {
+        try {
+          performReads(textIndex, mergedThreadWords, 10, 200, countDownLatch);
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      });
+
+      _threadPool.submit(() -> {
+        try {
+          performWrites(textIndex, firstThreadWords, 10, countDownLatch);
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      });
+
+      _threadPool.submit(() -> {
+        try {
+          performWrites(textIndex, secondThreadWords, 10, countDownLatch);
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      });
+
+      countDownLatch.await();
+    }
+
+    assertEquals(_resultSet.size(), mergedThreadWords.size());
+
+    assertTrue("ab not found in result set", _resultSet.contains("ab"));
```

Comment:
why not found?

Reply:
That's the message that will be shown if the element is not found